0000-interledger-tokens/0000-interledger-tokens.md

```diff
@@ -0,0 +1,242 @@
+
+## Background
+
+TODO
+
+## Motivation
+
+TODO
+
+## Terminology
+
+The following terms are used throughout this specification. Where they are used their meaning as follows:
+
+### Payer
+
+The entity whose funds are being spent when the token is used to initiate a payment.
+
+### Payer's Interledger Service Provider (ILSP)
+
+The entity that accepts and processes the token and payment request. This may be the payer or they may delegate it to some other entity that is able to make payments on their behalf.
+
+### Payee
+
+The entity that is requesting payment using the token.
+
+## The Interledger Token
+
+An **Interledger Token** is a special form of a JSON Web Token (JWT) [RFC7519](https://tools.ietf.org/html/rfc7519) that can be submitted to a participant on the Interledger network to request that they initiate a new payment via ILP. The content of the token describes how much to pay, who to pay and provides information for the entity processing the token to verify that the payment was authorized by the owner of the funds being sent.
+
+**Interledger Tokens** are limited profile of a standard JWT. This profile limits the allowed algorithms, allowed claims and defines specific rules for what constitutes valid claim values. A token is used by a payee to "claim" that a payment has been authorized by the payer in accordance with the terms of the token.
+
+After processing a token and making a payment the participant that has made the payment MAY return a new token to the payee to use again in the future. For example, a payee may posses a token that authorizes a payment to them of USD 100. They may then submit this to initiate a payment but request that only USD 25 is sent in the payment. Following successful execution of the payment they may be returned a new token that authorizes a payment to them of USD 75.
+
+An example token payload is shown below (with further details provided in later sections):
+
+```json
+{
+  "jti": "48359d89-e9ef-44dd-9a3d-68f991ed7556", //Unique Token ID
+  "iss": "872369652347412343", //Unique identifier for the payer (recognized by their ILSP)
+  "sub": "example.payee123", //ILP Address of the payee
+  "aud": "https://superwallet.example/ilp/", //Endpoint to which the token is submitted
+  "exp": "1518699103", //The Expiry of the token (Thursday, February 15, 2018 12:51:43 PM)
+  "payee": {
+    "max": "100", //The maximum amount that should be delivered to the payee when this token is presented 
+    "min": "0", //The minimum amount that should be delivered to the payee when this token is presented
+    "asset": "USD", //The asset that the min and max values are denominated in
+    "scale": "2" //The scale of the min and max (eg. if scale = 2 then a value of 1234 = USD 12.34)
```

Comment:
Relying on a fixed destination amount and currency is dangerous because there aren't really comparable assets in Interledger and the rate could change drastically between when you give out this token and when your ILSP goes to pay it. The payee could be running their own connector and mess with the rate to take all of your money.

Reply:
These are just a "guide". The payee wants to know what value the token has to them. The ILSP that does the sending should always respect the limits set in the payer data first.